Framework/Core/include/Framework/DataAllocator.h

```diff
@@ -91,12 +94,82 @@ class DataAllocator
   void
   adopt(const OutputSpec &spec, TObject*obj);
 
-  /// Serialize a snapshot of a TObject when called, which will then
+  /// Serialize a snapshot of a TObject when called, which will then be
   /// sent once the computation ends.
   /// Framework does not take ownership of the @a object. Changes to @a object
   /// after the call will not be sent.
-  void serializeSnapshot(const OutputSpec &spec, TObject* const object);
+  template <typename T>
+  typename std::enable_if<std::is_base_of<TObject, T>::value == true, void>::type
+  serializeSnapshot(const OutputSpec &spec, T *const object) {
+    std::string channel = matchDataHeader(spec, mRootContext->timeslice());
+    auto headerMessage = headerMessageFromSpec(spec, channel, o2::Header::gSerializationMethodROOT);
```

Comment:
wrong serialization method set?

Reply:
no, this is correct, no? if `is_base_of<TObject, T>` then we use root.